phpBB/phpbb/search/fulltext_sphinx.php

```diff
@@ -455,6 +465,49 @@ public function split_keywords(&$keywords, $terms)
 		return false;
 	}
 
+	/**
+	* Cleans search query passed into Sphinx search engine, as follows:
+	* 1. Hyphenated words are replaced with keyword search for either the exact phrase with spaces or as a single word without spaces eg search for "know-it-all" becomes ("know it all"|"knowitall*") 
+	* 2. Words with apostrophes are contracted eg "it's" becomes "its"
+	* 3. <, >, " and & are decoded from HTML entities.
+	* 4. Following special characters used as search operators in Sphinx are preserved when used with correct syntax:
+	* (a) quorum matching:  "the world is a wonderful place"/3 -- finds 3 of the words within the phrase. Number must be between 1 and 9.
+	* (b) proximity search: "hello world"~10 -- finds hello and world within 10 words of each other. Number can be between 1 and 99.
+	* (c) strict word order: aaa << bbb << ccc -- finds "aaa" only where it appears before "bbb" and only where "bbb" appears before "ccc".
+	* (d) exact match operator: if lemmatizer or stemming enabled, search will find exact match only and ignore other grammatical forms of the same word stem..
+	* eg raining =cats and =dogs -- will not return "raining cat and dog" 
+	* eg ="search this exact phrase" -- will not return "searched this exact phrase", "searching these exact phrases".
+	* 5. Special characters /, ~, << and = not complying with the correct syntax and other reserved operators are escaped and searched literally.
+	* Special characters not explicitly listed in charset_table or blend_chars in sphinx.conf will not be indexed and keywords containing them will be ignored by Sphinx.
+	* By default, only $, %, & and @ characters are indexed and searchable.
+	* String transformation is in backend only and not visible to the end user nor reflected in the results page URL or keyword highlighting.
+	*/
```

Comment:
Props on the well detailed docblock, makes a lot clear.
Some sentences are however a bit long and it's missing the `@param` and `@return` statement.

Perhaps something like this is a possibility:
```
/**
 * Clean search string.
 *
 * Cleans search query passed into Sphinx search engine, as follows:
 * 1. Hyphenated words are replaced with keyword search for either the exact phrase with spaces
 *    or as a single word without spaces eg search for "know-it-all" becomes ("know it all"|"knowitall*")
 * 2. Words with apostrophes are contracted eg "it's" becomes "its"
 * 3. <, >, " and & are decoded from HTML entities.
 * 4. Following special characters used as search operators in Sphinx are preserved when used with correct syntax:
 *    (a) quorum matching: "the world is a wonderful place"/3
 *        Finds 3 of the words within the phrase. Number must be between 1 and 9.
 *    (b) proximity search: "hello world"~10
 *        Finds hello and world within 10 words of each other. Number can be between 1 and 99.
 *    (c) strict word order: aaa << bbb << ccc
 *        Finds "aaa" only where it appears before "bbb" and only where "bbb" appears before "ccc".
 *    (d) exact match operator: if lemmatizer or stemming enabled,
 *        search will find exact match only and ignore other grammatical forms of the same word stem..
 *        eg. raining =cats and =dogs
 *            will not return "raining cat and dog"
 *        eg. ="search this exact phrase"
 *            will not return "searched this exact phrase", "searching these exact phrases".
 * 5. Special characters /, ~, << and = not complying with the correct syntax
 *    and other reserved operators are escaped and searched literally.
 *    Special characters not explicitly listed in charset_table or blend_chars in sphinx.conf
 *    will not be indexed and keywords containing them will be ignored by Sphinx.
 *    By default, only $, %, & and @ characters are indexed and searchable.
 *    String transformation is in backend only and not visible to the end user
 *    nor reflected in the results page URL or keyword highlighting.
 *
 * @param string	$search_string
 * @return string
 */
```

Reply:
Thank you. I thought it might be a bit wordy but I wanted to be transparent about what I was trying to achieve. I will take up all of your suggestions and make some further changes.